### PR TITLE
feat: Implement Query in the liaison

### PR DESCRIPTION
**Pull request by Fine0830**
Implement Query in the liaison

Signed-off-by: Qiuxia Fan <qiuxiafan@apache.org>

### CODE REVIEW

File: banyand/liaison/grpc/grpc.go [hunk 1]
```diff
@@ -334,9 +335,30 @@ func (s *Server) Write(TraceWriteServer v1.TraceService_WriteServer) error {
...
+		return nil, errQuery
+	}
+	msg, errFeat := feat.Get()
+	if errFeat != nil {
+		return nil, errFeat
+	}
+	queryMsg := msg.Data()
+
+	var arr []*v1.Entity
+	switch reflect.TypeOf(queryMsg).Kind() {
```
Comment (lujiajing1126) [inline banyand/liaison/grpc/grpc.go:350]: You actually do not need reflect here.

Just use [type-assertion](https://tour.golang.org/methods/15), see the examples.

File: banyand/liaison/grpc/grpc_test.go [hunk 2]
```diff
@@ -234,13 +233,11 @@ func traceQuery(t *testing.T, conn *grpclib.ClientConn) {
...
-		Projection("http.method", "service_id", "service_instance_id").
-		Fields("service_id", "=", "my_app", "http.method", "=", "GET").
+		Limit(10).
+		Offset(0).
+		Metadata("default", "sw").
 		TimeRange(sT, eT).
+		Projection("trace_id").
 		Build()
 	stream, errRev := client.Query(ctx, criteria)
 	if errRev != nil {
```
Comment (hanahmily) [inline banyand/liaison/grpc/grpc_test.go:243]: The query result should match the written data.

**Review by hanahmily — CHANGES_REQUESTED** [review]
The test case should check the query result matches the expectation.

**Comment by Fine0830** [comment]
Done

**Review by hanahmily — APPROVED** [review]
LGTM

**Comment by kezhenxu94** [comment]
The test seems to be flaky

https://github.com/apache/skywalking-banyandb/pull/38/checks?check_run_id=3346386041#step:5:76

**Pull request merged**